### PR TITLE
feat: [EXT-2242] Add typings and expose metadata methods

### PR DESCRIPTION
**Pull request by colomolo**
# Purpose of PR
Add proper typings for a metadata of entry, return `getMetadata` and `onMetadataChange` methods at `createEntry` resulting object.

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -98,6 +98,10 @@ export interface EntrySys {
   contentType: Link
 }
 
+export type Metadata = {
+  tags?: Link[]
+} | undefined
```
Comment (andipaetzold) [inline lib/types.ts:103]: I think it's better to add `undefined` to the `getMetadata` and `onMetadataChanged` types below. That way we can properly use `Metadata` as a type when we know that it is set.

File: lib/types.ts [hunk 2]
```diff
@@ -227,9 +231,9 @@ export interface EntryAPI extends TaskAPI {
    * Optional metadata on an entry
    * @deprecated
    */
-  metadata?: {
-    tags?: Link[]
-  }
+  metadata?: Metadata,
+  getMetadata: () => Metadata,
+  onMetadataChanged: (callback: (metadata: Metadata) => void) => Function,
```
Comment (andipaetzold) [inline lib/types.ts:236]:
```suggestion
  metadata?: Metadata
  getMetadata: () => Metadata | undefined
  onMetadataChanged: (callback: (metadata?: Metadata) => void) => () => void
```

**Review by andipaetzold — CHANGES_REQUESTED** [review]
Already looks good. Just some type changes.

File: lib/types.ts (hunk 1 above)
Comment (colomolo) [reply lib/types.ts]: Yeah, thought about the same and fixed it.

File: lib/types.ts [hunk 3]
```diff
@@ -231,9 +231,9 @@ export interface EntryAPI extends TaskAPI {
    * Optional metadata on an entry
    * @deprecated
    */
-  metadata?: Metadata,
-  getMetadata: () => Metadata,
-  onMetadataChanged: (callback: (metadata: Metadata) => void) => Function,
+  metadata?: Metadata | undefined
```
Comment (andipaetzold) [inline lib/types.ts:234]: nit: `metadata?: Metadata` is actually the same as `metadata?: Metadata | undefined`

**Comment by andipaetzold** [comment]
Looks good - gonna approve after testing with a real app locally

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**